sdk/core/Azure.Core/src/Shared/EventSourceEventFormatting.cs

```diff
@@ -18,12 +19,24 @@ public static string Format(EventWrittenEventArgs eventData)
 
             if (eventData.Message != null)
             {
-                return string.Format(CultureInfo.InvariantCulture, eventData.Message, payloadArray);
+                try
+                {
+                    return string.Format(CultureInfo.InvariantCulture, eventData.Message, payloadArray);
+                }
+                catch (FormatException)
```

Comment:
in some occasions the message inside the eventData is not formattable.
when this happend the program recieves a FormatException in this point.

solution,
catch the format exception and build a message without the formatting.